### DISCUSSION TITLE
Vitest not mocking dependency when module uses it?

### DISCUSSION THREAD

**Discussion by RPDeshaies** [vitest-dev/vitest#587] [Q&A]
Hi,

I was wondering if it was normal that Vitest doesn't use the provided module mock when a package listed as a dependency is using it ?

What I mean is, we have an `internal` package (bundled in CommonJS) that uses `dependency-to-mock`.

When I call 

```
vi.mock("dependency-to-mock", ()=> { 
// return my mock
})
```

I expect Vitest to mock all imports of that module, but I have the impression it only works if **my** code imports `dependency-to-mock`, but it doesn't work if another package does it.

Any thoughts? Thanks in advance! :)

**Comment by sheremet-va** [comment]
Yes, It works only for code, that goes through vite-node pipeline. Usually it's user code, but you can try inlining it with `deps.inline`

**Reply by RPDeshaies** [reply]
So I'd do something like 

```
test : {
  deps : {
    inline:["dependency-to-mock"] // or would I pass the name of the lib (in my example "internal") here ?
  }
}
```

**Reply by sheremet-va** [reply]
You need to inline deps that imports the mocked module

**Reply by RPDeshaies** [reply]
In this case, I get something like

```
 > node_modules/my-internal-library/file.js:7:23: error: Could not resolve "dependency-tomock" (mark it as external to exclude it from the bundle, or surround it with try/catch to handle the failure at run-time)
```

Any other suggestions? Right now, I'm thinking about :  

- Exporting the content of my module inside a single file (export * from 'internal module')
- Updating all my files to load from this file instead of the module directly
- Telling Vitest to mock that file (that way the original import just never happens)

Expect if you have a better idea? (thanks for helping me BTW, it's super appreciated)
